test/tensorflow_test.py

```diff
@@ -187,5 +187,20 @@ def testNetworkDriverWorkerIndependent(self):
 
     ray.worker.cleanup()
 
+  def testVariablesControlDependencies(self):
+    ray.init(num_workers=1)
+
+    # Creates a network and appends a momentum optimizer.
+    sess = tf.Session()
+    loss, _ = make_linear_network()
+    minimizer = tf.train.MomentumOptimizer(0.9, 0.9).minimize(loss)
+    net_vars = ray.experimental.TensorFlowVariables(minimizer, sess)
+   
+    # Tests if all variables are properly retrieved, 2 variables and 2 momentum
+    # variables.
+    self.assertEqual(len(net_vars.variables.items()), 4)
+
```

Comment:
Can you also call `get_weights` and `set_weights` and make sure those methods work in this case? (you'll probably need to call init as well).